### ISSUE TITLE
✨ Basic contrast limits UI element is too abstract.

### ISSUE THREAD

**Issue by d-v-b** [napari/napari#3347]
### Overview of design need

(related to #3346)

Augmenting the contrast limits slider with a histogram would be _super_ helpful. It could show either the distribution of the values or the cumulative distribution of the values (neuroglancer-style, which I've come to enjoy). 

Also, I think it should be a priority to show the user _actual numbers_ alongside the abstract representation of the interval. Unless I'm missing something, to associate actual numbers with the contrast adjustment interval I need to access a subtle extra menu via right click? (Which triggers a modal dialog, so I can't even alt-tab to my screenshot software to take a picture of it easily :cry:)

### What level of design is needed? (Choose all that apply)
_This section may be updated by the designer / UX researcher working on this issue_

- [x] **User experience research:**  high-level recommendation/exploration of user needs, design heuristics, and / or best practices to inform a design experience  
(Use this option when you feel there’s a challenge to be solved, but you’re curious about what the experience should be — may involve research studies to understand challenges/opportunities for design)
- [x] **information flow / conceptual:** organizing and structuring of information flow and content, including layout on screen or across multiple steps
- [x] **Visual:** creating mockups, icons, etc
(If choosing this level alone, it means that the content to be mocked up and its organization is already known and specified)

### Is design a blocker?
- [x] **Yes:** engineering cannot proceed without a design first
- [ ] **No:** engineering can create a first version, and design can come in later to iterate and refine

If selecting **Yes**, how much design input is needed to unblock engineering? For example, is a full, final visual design needed, or just a recommendation of which conceptual direction to go?

**Comment by tlambert03** [comment]
duplicate of https://github.com/napari/napari/pull/675 ... histograms are still a high priority, don't worry.  there's been a lot of discussion about general "plotting" inside of napari, so we're hesitant to introduce new frameworks for something like a histogram without considering the broader usage of plots.

**Comment by liaprins-czi** [comment]
If this is truly a dup, should it be closed? Just want to verify that #675 covers what this issue is bringing up. Also wholeheartedly agree about being careful with introducing "new frameworks for something like a histogram without considering the broader usage of plots"!

**Comment by d-v-b** [comment]
A histogram would be great, but there's stuff that could be done without one.

The GUI represents the contrast limits as an abstract interval. 
![image](https://user-images.githubusercontent.com/3805136/132400189-4cd97134-9fee-469a-a19c-c2e07d7650f6.png)
As a user, it feels very "form over function". I'm personally familiar with the minimalist design philosophy that inspired napari :wink: but I think it comes at a cost here.

 - This UI element doesn't get updated when the contrast limits change. i.e., a user pressing "once" or "continuous" gets no feedback that their input did anything. This was the original impetus for this issue, actually. I was pressing these buttons and seeing no effect anywhere in the GUI, which made me wonder if they were doing anything. The actual numbers of the min and max are invisible (as are opacity and gamma). This makes it harder to use the software repeatably ("move the gamma slider to roughly 1/3 of the way up" vs "set gamma to .3"). This can be solved by making space in the layer controls element for the min and max value of the contrast limits, as well as the start and end of the interval. If I saw numbers changing in the layer control panel when I clicked "once" or "continuous" as I scrolled through my data, it would be more clear what those UI elements were doing.

- The "barbell" control for the interval has weird behavior at the limits:  
![image](https://user-images.githubusercontent.com/3805136/132400959-351ccda8-b02d-40e9-8909-0bc0d5175b64.png)
If you drag all the way to one side, it looks like a single point slider, and it becomes hard to interact with the element. This can be solved by styling the barbell with an outline, or just ditching the barbell. I'm not sure that a medium-sized circle is the best way to represent a point on an interval. A lollipop might be better if you still want to have a circle involved, but with the visual precision of a line.

**Comment by tlambert03** [comment]
> This UI element doesn't get updated when the contrast limits change

can you clarify what you mean here?  we certainly intend for it to change, and it does for me:


https://user-images.githubusercontent.com/1609449/132404544-a085f967-c1b0-4b63-ba6d-5dd21a5aa1b2.mov

**Comment by d-v-b** [comment]
OK, I think I understand why I didn't see this behavior -- my data was already normalized, so "once" and "continuous" did nothing to the contrast limits. And if the button can trigger an idempotent action (as this one should), then the user can't expect to see the gui change :)

**Comment by tlambert03** [comment]
hah :)  yep... autoscaling pre-normalized data is unlikely to show anything.

**Comment by d-v-b** [comment]
that being said, if I saw the actual numbers for the contrast limits displayed, then I would know for sure that pressing the "once" button didn't do anything. If I only see an abstract interval slider with no numbers, I can't be sure that my button press didn't alter the definition of the endpoints of the slider. Showing numbers makes it unambiguous.

**Comment by tlambert03** [comment]
absolutely, we all agree that we need some more heads up display on the underlying numbers in the image.  A histogram would do it, some additional stats would do it.  We're definitely not "done" when it comes to showing numeric detail...  I do think there's hesitancy (with good reason) to just throw numbers somewhere without carefully considering the space available.  I know it's taken longer than many might like, but rest assured it's something we all agree needs improvement

**Comment by tlambert03** [comment]
btw, when we added labeled sliders in #2753, this came up and we specifically left out the contrast limits slider because it basically runs out of room immediately if you have like a 16 bit image.  @liaprins-czi, if you'd like to help us come up with a way to show numbers (similar to what is shown when you right click on the slider, but without right clicking), without cramping the slider... that would be great.

Maybe we add a simple text readout below? the slider? (prettier than this of course... but you get the idea)
![Screen Shot 2021-09-07 at 4 22 34 PM](https://user-images.githubusercontent.com/1609449/132406340-c55edb01-4af1-4b29-89de-0928f683cc60.png)

**Comment by liaprins-czi** [comment]
Yes this will fit into the design of the layer controls area in general
